client/src/pages/Login.js

```diff
@@ -210,6 +210,11 @@ const Login = ({ isLoginForm, forgotPassword, isAuthenticated }) => {
   const code = queryParams.get("code");
   const state = queryParams.get("state");
 
+  useEffect(() => {
+    //allows user to return to previous page by preventing unwanted redirects
+    sessionStorage.removeItem("createPostAttemptLoggedOut");
+  }, []);
+
```

Comment:
But after this, if a user tries to create a post while logged out, the modal will not automatically open after login.
I would suggest adding this to the Home page, so when the user goes back there by clicking on the logo, it will be cleared, and they can go back to feed.

Reply:
@Zaydme new commit should allow for the modal window to show after login and prevent unwanted redirect or needing to return to the homepage.